app/interactors/notify_about_checkin.rb

```diff
@@ -5,6 +5,30 @@ class NotifyAboutCheckin
 
   def call
     device.notify_subscribers("new_checkin", checkin)
-    device.broadcast_checkin_for_friends(checkin)
+    broadcast_checkin_for_friends
+  end
+
+  private
+
+  def broadcast_checkin_for_friends
+    device.user.friends.find_each do |friend|
+      next unless ConnectedList.all.include?(friend.id.to_s) && broadcast_checkin?(friend)
```

Comment:
Without the ConnectedList check it will have to do all the permission checks to get the correct check-in for no reason. Considering this happens every time you check-in, wanted to make sure we were only doing work here if it was absolutely necessary.